grades/models.py

```diff
@@ -231,6 +232,15 @@ def for_user_course(cls, user, course):
         now = now_in_utc()
         return cls.objects.filter(user=user, course=course, exam_run__date_grades_available__lte=now)
 
+    def set_score(self, new_score):
+        """
+        Sets a value for the score field, then sets related fields based on that value
+        """
+        self.score = new_score or 0.0
```

Comment:
This doesn't seem right to me. If we get a falsy score we should probably error instead of defaulting to 0.